### ISSUE TITLE
Latest Preview 1.0 - beta4 - Enum inferred type is int 

### ISSUE THREAD

**Issue by jkone27** [fsprojects/SwaggerProvider#186]
### Description

https://swagger.io/docs/specification/data-models/enums/

```json
 {
            "name": "target",
            "in": "query",
            "description": "Service to search. Leave blank or specify 'undefined' to search globally, specify any other target to search just that section of data.",
            "required": false,
            "type": "string",
            "enum": [ "Undefined", "CustomerName", "CustomerAddress", "CustomerEmail", "OrderNumber", "CustomerPaymentProviderReference", "CustomerPaymentInternalReference", "FlightProductCheckinAddress", "Pnr", "TicketNumber", "CustomerCompany", "InsurancePolicyNumber", "SupplierReference" ]
}
```
and SwaggerProvider generated type instead is **int**, wheras should be string (or eventually an Enum with valid values, before it was string with the old "LTS" package

This causes new generate types not to be compatible with "old" ones.

![image](https://user-images.githubusercontent.com/10419217/121780434-93371a80-cba0-11eb-89be-342ce355717d.png)

![image](https://user-images.githubusercontent.com/10419217/121780485-d2fe0200-cba0-11eb-8b49-80714ad41a02.png)

**Comment by Thorium** [comment]
I have the same issue, schema:
"country_code":{"enum":["Unknown","CAN","USA"],"type":"string"}

... with SwaggerProwider 0.12 (NewtonSoft serialization) it used to serialize to `String` properly.
After SwaggerProwider 1.0 (System.Text.Json serialization) it serializes incorrectly to `Option<Int>`.

**Comment by sergey-tihon** [comment]
@Thorium 0.12.2 is release - https://www.nuget.org/packages/SwaggerProvider/0.12.2 and thanks for the fix.

@jkone27 can you share the full schema with the issue and code snippet (what TP you used? SwaggerProvider or OpenApiProvider?)

**Comment by Thorium** [comment]
Here is a random schema for SwaggerProvider that has the enum-issue:
https://api.sandbox.repayonline.com/rgapi/v1/swagger
https://api.sandbox.repayonline.com/rgapi/swagger/ui/index

**Comment by jkone27** [comment]
I used OpenApiProvider, dont have the schema with me rn but i hope @Thorium is as good to reproduce

**Comment by sergey-tihon** [comment]
Should be fixed in [SwaggerProvider 2.0.0-beta2](https://www.nuget.org/packages/SwaggerProvider/2.0.0-beta2)
